src/main/java/gift/external/kakao/KakaoApiClient.java

```diff
@@ -0,0 +1,91 @@
+package gift.external.kakao;
+
+import gift.common.exception.BusinessException;
+import gift.common.exception.code.ExternalErrorCode;
+import gift.external.kakao.request.GetTokenRequestDto;
+import gift.external.kakao.request.RefreshRequestDto;
+import gift.external.kakao.response.GetMemberIdResponseDto;
+import gift.external.kakao.response.GetTokenResponseDto;
+import gift.external.kakao.response.RefreshResponseDto;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.HttpServerErrorException;
+import org.springframework.web.client.RestClient;
+import org.springframework.web.client.RestClientException;
+
+@Component
+public class KakaoApiClient {
+
+    private static final Logger log = LoggerFactory.getLogger(KakaoApiClient.class);
+    private RestClient.Builder builder;
+
+    @Value("${kakao.client-id}")
+    private String clientId;
+
+    public KakaoApiClient(RestClient.Builder builder) {
+        this.builder = builder;
+    }
```

Comment:
```suggestion
    public KakaoApiClient(RestClient.Builder builder) {
        this.kauthClient = builder.baseUrl("https://kauth.kakao.com").build();
        this.kapiClient = builder.baseUrl("https://kapi.kakao.com").build();
    }
```
baseUrl은 공통값이라 이런식으로 초기화해줘도 좋겠네요😊